lib/semantics/scope.cc

```diff
@@ -64,6 +74,63 @@ DerivedTypeSpec &Scope::MakeDerivedTypeSpec(const SourceName &name) {
   return derivedTypeSpecs_.back();
 }
 
+Scope::ImportKind Scope::importKind() const {
+  if (importKind_) {
+    return *importKind_;
+  }
+  if (symbol_) {
+    if (auto *details{symbol_->detailsIf<SubprogramDetails>()}) {
+      if (details->isInterface()) {
+        return ImportKind::None;  // default for interface body
+      }
+    }
+  }
+  return ImportKind::Default;
+}
+
+std::optional<parser::MessageFixedText> Scope::set_importKind(ImportKind kind) {
+  if (!importKind_.has_value()) {
+    importKind_ = kind;
+    return std::nullopt;
+  }
+  std::optional<parser::MessageFixedText> error;
+  bool hasNone{kind == ImportKind::None || *importKind_ == ImportKind::None};
+  bool hasAll{kind == ImportKind::All || *importKind_ == ImportKind::All};
+  // Check C8100 and C898
```

Comment:
A brief summary of these two constraints would be great, if they're not too complicated.

Reply:
OK